docs/source/explanation/singleuser.md

```diff
@@ -0,0 +1,108 @@
+(singleuser)=
+
+# The JupyterHub single-user server
+
+When a user logs into JupyterHub, they get a 'server', which we usually call the **single-user server**, because it's a server that's meant for a single JupyterHub user.
+Each JupyterHub user gets a different one (or more than one!).
+
+A single-user server is a process running somewhere that is:
+
+1. accessible over http[s],
+2. authenticated via JupyterHub OAuth,
+3. started by a [Spawner](spawners), and
+4. 'owned' by a single JupyterHub user
+
+(singleuser-auth)=
+
+## Single-user server authentication
+
+Implementation-wise, JupyterHub single-user servers are a special-case of [](services),
+and as such use the same (oauth2) authentication mechanism (more on oauth in JupyterHub at [](oauth)).
+This is primarily implemented in the {class}`~.HubOAuth` class.
+
+The default single-user command, `jupyterhub-singleuser`, launches `jupyter-server`, the same program used when you run `jupyter lab` on your laptop.
+(_It can also launch the legacy `jupyter-notebook` server_).
+That's why JupyterHub looks familiar to folks who are already using Jupyter at home or elsewhere.
+It's the same!
+`jupyterhub-singleuser` _customizes_ that program to change (approximately) one thing: **authenticate requests with JupyterHub**.
+
+`jupyter-server` allows customizing _how_ a user is authenticated when making requests to the server.
+By default, `jupyter-server` uses its own cookie to authenticate.
+If that cookie is not present, the server redirects you a login page and asks you to enter a password or token.
+
+This code resides in `jupyterhub.singleuser` subpackage of JupyterHub.
+The main task of this code is to:
```

Comment:
The transition between the last two paragraph is a bit vague by switching from speaking of "password or token" to "This code". I got thinking that "This code" referred to the password or token, and that the cookie was named "jupyterhub.singleuser" or similar.

I think maybe the last paragraph was meant to be written directly after line 27 initially and now has a "this" that is a bit disconnected? I think replacing "this code" with an explicit "the `jupyterhub-singleuser` code".

Reply:
Yup, this is my brain branching off several narrative threads, which I struggle to represent as a linear document. I tried to reflow it a bit.